### PR TITLE
Restore documentation that had been moved to zcash.readthedocs.io

### PR DESCRIPTION
**Pull request by nuttycom**
This reverts https://github.com/zcash/zcash/commit/70ebc0942af257a72060165579b3f4b44ecefb06 which
had previously moved zcashd documentation to zcash.readthedocs.io, backports changes that had
been made on the readthedocs site, and moves the update documentation into the zcashd book.

### CODE REVIEW

File: doc/dnsseed-policy.md [hunk 1]
```diff
@@ -1,54 +1,53 @@
 Expectations for DNS Seed operators
 ====================================
 
-Zcash attempts to minimize the level of trust in DNS seeds,
-but DNS seeds still pose a small amount of risk for the network.
-As such, DNS seeds must be run by entities which have some minimum
-level of trust within the Zcash community.
+Zcash attempts to minimize the level of trust in DNS seeds, but DNS seeds still
```
Comment (str4d) [inline doc/dnsseed-policy.md]: This is a document from upstream, so I would prefer we don't reformat it in this way (for the same reason we don't reformat the entire codebase), unless doing so is required for correct book rendering.

File: doc/wallet-backup.md [hunk 2]
```diff
@@ -1,91 +1,146 @@
-# Wallet Backup Instructions
+Wallet Backup Instructions
+==========================
```
Comment (str4d) [inline doc/wallet-backup.md]: These changes should not be included; they are changing the documentation from Markdown to rST format (which is what the external repo required), but here we _want_ Markdown.

File: doc/tor.md [hunk 3]
```diff
@@ -1,149 +1,289 @@
-*** Warning: Do not assume Tor support does the correct thing in Zcash; better Tor support is a future feature goal. ***
-
-TOR SUPPORT IN ZCASH
+Tor Support in Zcash
 ====================
 
-It is possible to run Zcash as a Tor hidden service, and connect to such services.
```
Comment (str4d) [inline doc/tor.md]: This document is originally from upstream, and this rewrite is definitely going to cause backporting fun. I'd prefer that changes to documents from upstream be done in a separate commit, so it's easier to isolate during backporting.

**Comment by mdr0id** [comment]
I'm not quite sure I understand this PR. Are now going to support 2 sets of user style guides in read the docs and in this repo again?

**Comment by skyl** [comment]
> I'm not quite sure I understand this PR. Are we now going to support 2 sets of user style guides in read the docs and in this repo again?

Would it make sense to deprecate the RTD repo and just push the RTD site as an artifact from here? I was looking to get started on zcash/zcash today to look at the speed of the tests and I got sidetracked looking for the RTD source to fix a few small things there (it says 4.6.0 and says `rpc-tests.sh` instead of `rpc-tests.py`, for example). If the RTD source was in this repo, it would have been a better experience.

**Comment by nuttycom** [comment]
> I'm not quite sure I understand this PR. Are we now going to support 2 sets of user style guides in read the docs and in this repo again?

The intent would be for everything that is specifically zcashd related to be removed from the readthedocs site, and for it to point to the zcashd book for zcashd-specific documentation.

File: doc/dnsseed-policy.md (hunk 1 above)
Comment (sellout) [reply doc/dnsseed-policy.md]: Undid this change.

File: doc/tor.md (hunk 3 above)
Comment (sellout) [reply doc/tor.md]: This change is now in its own commit.

File: doc/files.md [hunk 4]
```diff
@@ -1,14 +1,23 @@
...
-* debug.log: contains debug information and general logging generated by zcashd
-* fee_estimates.dat: stores statistics used to estimate minimum transaction fees and priorities required for confirmation
-* peers.dat: peer IP address database (custom format)
-* wallet.dat: personal wallet (BDB) with keys and transactions
-* .cookie: session RPC authentication cookie (written at start when cookie authentication is used, deleted on shutdown): since 0.12.0
-* onion_private_key: cached Tor hidden service private key for `-listenonion`: since 0.12.0
+Data Directory Files
+====================
+
+Files within the zcashd data directory (``~/.zcash/``) on Linux unless otherwise specified) include:
```
Comment (daira) [inline doc/files.md]:
```suggestion
Files within the zcashd data directory (``~/.zcash/`` on Linux unless otherwise specified)
include:
```

File: doc/security-warnings.md [hunk 5]
```diff
@@ -9,12 +9,12 @@ announcements, audit results and other general security information, see
...
+-----------------
 
-There are [known bugs](https://github.com/scipr-lab/libsnark/issues/26) which
-make proving keys generated on 64-bit systems unusable on 32-bit and big-endian
-systems. It's unclear if a warning will be issued in this case, or if the
-proving system will be silently compromised.
+There are [known bugs](https://github.com/scipr-lab/libsnark/issues/26/)
+which make proving keys generated on 64-bit systems unusable on 32-bit and
+big-endian systems. It's unclear if a warning will be issued in this case, or
+if the proving system will be silently compromised.
```
Comment (daira) [inline doc/security-warnings.md]:
```suggestion
```

File: doc/book/src/user/security-warnings.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+https://z.cash/support/security.html
+
+x86-64 Linux Only
+-----------------
+
+There are [known bugs](https://github.com/scipr-lab/libsnark/issues/26/)
+which make proving keys generated on 64-bit systems unusable on 32-bit and
+big-endian systems. It's unclear if a warning will be issued in this case, or
+if the proving system will be silently compromised.
+
```
Comment (daira) [inline doc/book/src/user/security-warnings.md:18]:
```suggestion
```

File: doc/wallet-backup.md [hunk 7]
```diff
@@ -2,90 +2,142 @@
...
+This will generate a file in the export directory listing all transparent and
+shielded private keys with their associated public addresses. The file path
+will be returned in the command line.
 
 To import keys into a wallet which were previously exported to a file, use:
 
-`zcash-cli z_importwallet <path/to/exportdir/nameofbackup>`
+```bash
+$ zcash-cli z_importwallet <path/to/exportdir/nameofbackup>
+```
```
Comment (daira) [inline doc/wallet-backup.md]: This need work. HD-generated keys won't be included in the export file and therefore won't be imported; this includes all Sapling and Orchard keys generated since and including zcashd 4.7.0.

File: doc/wallet-backup.md [hunk 8]
```diff
@@ -2,90 +2,142 @@
...
-See the command's help documentation for instructions on fine-tuning the wallet rescan:
+See the command's help documentation for instructions on fine-tuning the wallet
+rescan:
 
-`zcash-cli help importprivkey`
+```bash
+$ zcash-cli help importprivkey
+```
 
 ### Using `dumpwallet`
```
Comment (daira) [inline doc/wallet-backup.md]: This is currently disabled by default.

File: doc/tor.md [hunk 9]
```diff
@@ -1,149 +1,289 @@
...
+#HiddenServicePort 8233 127.0.0.1:8233
+
+# use the generated v2 Onion hostname until v3 support is complete
+HiddenServiceDir /var/lib/tor/zcash_hidden_service_v2/
+HiddenServiceVersion 2
+HiddenServicePort 8233 127.0.0.1:8233
+```
+
+Note that zcashd does not yet support Onion v3 addresses, but will do so before
+v2 addresses are removed from Tor.  See [this
```
Comment (daira) [inline doc/tor.md]: Nope.

**Review by daira — APPROVED** [review]
ACK with comments.

File: doc/book/src/user/files.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
+Data Directory Files
+====================
+
+Files within the zcashd data directory (``~/.zcash/``) on Linux unless otherwise specified) include:
```
Comment (daira) [inline doc/book/src/user/files.md:4]:
```suggestion
Files within the zcashd data directory (``~/.zcash/`` on Linux unless otherwise specified)
include:
```

File: doc/book/src/user/security-warnings.md [hunk 11]
```diff
@@ -75,6 +67,9 @@ The REST interface is a feature inherited from upstream Bitcoin.  By default,
 it is disabled. We do not recommend you enable it until it has undergone a
 security review.
 
+RPC Interface
+---------------
```
Comment (daira) [inline doc/book/src/user/security-warnings.md:71]:
```suggestion
-------------
```

**Review by daira — APPROVED** [review]
ACK

**Review by str4d — APPROVED** [review]
ACK. There is more we should do with these, but we can do that in subsequent PRs once we've seen how it fits into the book.
